### PR TITLE
AND-4910 Added NEAR address validation to release 5.0.

### PR DESCRIPTION
**Pull request by Sateetas**

### CODE REVIEW

File: app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressMiddleware.kt [hunk 1]
```diff
@@ -54,20 +59,52 @@ internal class AddressMiddleware {
...
     private fun verifyAddress(
+        address: String,
+        walletManager: WalletManager,
+        sourceType: AddressEntered.SourceType?,
+        isUserInput: Boolean,
+        dispatch: (Action) -> Unit,
+    ) {
+        val wallet = walletManager.wallet
+
+        scope.launch(Dispatchers.Main) {
```
Comment (Mama1emon) [inline app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressMiddleware.kt]: `mainScope` для упрощения

File: app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressValidator.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import com.tangem.blockchain.blockchains.near.NearWalletManager
+import com.tangem.blockchain.blockchains.near.network.NearAccount
+import com.tangem.blockchain.common.Blockchain
+import com.tangem.blockchain.common.Wallet
+import com.tangem.blockchain.common.WalletManager
+import com.tangem.blockchain.extensions.Result
+import com.tangem.tap.features.send.redux.AddressVerifyAction
+
+class AddressValidator {
```
Comment (Mama1emon) [inline app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressValidator.kt]: `internal`

File: app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressValidator.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            } else {
+                AddressVerifyAction.Error.ADDRESS_SAME_AS_WALLET
+            }
+        } else {
+            AddressVerifyAction.Error.ADDRESS_INVALID_OR_UNSUPPORTED_BY_BLOCKCHAIN
+        }
+    }
+
+    companion object {
+        const val NEAR_IMPLICIT_ADDRESS_LENGTH = 64
```
Comment (Mama1emon) [inline app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressValidator.kt]: `private`

File: app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressMiddleware.kt (hunk 1 above)
Comment (Sateetas) [reply app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressMiddleware.kt]: ok

File: app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressValidator.kt (hunk 2 above)
Comment (Sateetas) [reply app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressValidator.kt]: ok

File: app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressValidator.kt (hunk 3 above)
Comment (Sateetas) [reply app/src/main/java/com/tangem/tap/features/send/redux/middlewares/AddressValidator.kt]: ok

**Review by Mama1emon — APPROVED** [review]

**Review by kozarezvlad — APPROVED** [review]

**Pull request merged**